servicex/resources/submit_transformation_request.py

```diff
@@ -160,6 +160,10 @@ def post(self):
                 request_rec.generated_code_cm = \
                     self.code_gen_service.generate_code_for_selection(request_rec, namespace)
 
+            # Insure the required queues and exchange exist in RabbitMQ broker
+            self.rabbitmq_adaptor.setup_exchange('transformation_requests')
+            self.rabbitmq_adaptor.setup_exchange('transformation_failures')
```

Comment:
Note that if RabbitMQ is unreachable, these lines will throw an exception with the following traceback:
```
  File "/home/servicex/servicex/resources/submit_transformation_request.py", line 164, in post
    self.rabbitmq_adaptor.setup_exchange('transformation_requests')
  File "/home/servicex/servicex/rabbit_adaptor.py", line 97, in setup_exchange
    channel = self.channel
  File "/home/servicex/servicex/rabbit_adaptor.py", line 78, in channel
    self.connect()
  File "/home/servicex/servicex/rabbit_adaptor.py", line 57, in connect
    self._connection = pika.BlockingConnection(self._url_list)
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/blocking_connection.py", line 360, in __init__
    self._impl = self._create_connection(parameters, _impl_class)
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/blocking_connection.py", line 451, in _create_connection
    raise self._reap_last_connection_workflow_error(error)
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/utils/selector_ioloop_adapter.py", line 567, in _resolve
    self._flags)
  File "/usr/local/lib/python3.6/socket.py", line 745, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
[Errno -2] Name or service not known
```
This is an instance of `socket.gaierror`, a pretty low level networking exception. The response will be 500 Internal Server Error, which is okay, but it might be nice to catch it and return [`503 Service Unavailable`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503).

Reply:
Yes - that's much better